packages/backend/src/routers/projectRouter.ts

```diff
@@ -700,6 +700,74 @@ projectRouter.post(
     },
 );
 
+projectRouter.post(
+    '/sqlRunner/downloadCsv',
+    allowApiKeyAuthentication,
+    isAuthenticated,
+    async (req, res, next) => {
+        try {
+            const results: ApiSqlQueryResults =
+                await projectService.runSqlQuery(
+                    req.user!,
+                    req.params.projectUuid,
+                    req.body.sql,
+                );
+
+            const csvHeader = Object.keys(results.rows[0]);
```

Comment:
Now we use this logic in 3 different places. We should unify this code; I've recently created CsvService (check convertApiResultsToCsv) , but I haven't refactored the existing code. But it can be done on a separate PR

Reply:
I'll open an issue for this

Reply:
https://github.com/lightdash/lightdash/issues/4543